source/code/core/core/public/ice/base.hxx

```diff
@@ -39,3 +39,18 @@ namespace ice
     using clean_type = clear_type_t<T>;
 
 } // namespace ice
+
+//! \brief This macro is required for a bug apprearing in the MSVC compile when generating optimized code with /O2
```

Comment:
This issue was also observed in other locations. Maybe the coroutine API is not properly defined?

Reply:
https://github.com/iceshard-engine/engine/issues/111